metricflow/test/model/validations/test_data_warehouse_tasks.py

```diff
@@ -0,0 +1,68 @@
+from metricflow.model.data_warehouse_model_validator import (
+    DataWarehouseModelValidator,
+    DataWarehouseTaskBuilder,
+    DataWarehouseValidationTask,
+)
+
+from metricflow.model.objects.data_source import DataSource, Mutability, MutabilityType
+from metricflow.model.objects.elements.dimension import Dimension, DimensionType
+from metricflow.model.objects.user_configured_model import UserConfiguredModel
+from metricflow.model.validations.validator_helpers import ValidationIssueLevel
+from metricflow.protocols.sql_client import SqlClient
+from metricflow.specs import DimensionReference
+
+
+def test_build_data_source_tasks() -> None:  # noqa:D
+    dim_ref_1 = DimensionReference(element_name="dim1")
+    dim_ref_2 = DimensionReference(element_name="dim1")
+
+    model = UserConfiguredModel(
+        data_sources=[
+            DataSource(
+                name="test_data_source",
+                sql_table="test.data_source",
+                dimensions=[
+                    Dimension(name=dim_ref_1, type=DimensionType.CATEGORICAL, is_partition=True),
+                    Dimension(
+                        name=dim_ref_2,
+                        type=DimensionType.CATEGORICAL,
+                        is_partition=True,
+                        expr="animal in ('cat', 'dog', 'platypus')",
+                    ),
+                ],
+                mutability=Mutability(type=MutabilityType.IMMUTABLE),
+            ),
+        ],
+        metrics=[],
+        materializations=[],
+    )
+
+    tasks = DataWarehouseTaskBuilder.gen_data_source_tasks(model=model)
```

Comment:
Can you also add a call for `validate_data_sources()`?

Reply:
Part of me thinks that having a `validate_data_sources` test doesn't actually add value. Currently `validate_data_sources` is two lines, a call to `gen_data_source_tasks` and a call to `run_tasks` with those returned tasks. And both of these methods are currently tested. However, I guess a test specifically for `validate_data_sources` ensures that changes directly to `validate_data_sources` doesn't break it 🤷

Reply:
It - is `validate_data_sources` the main method that clients are expected to use?

Reply:
The `validate_data_sources` method is essentially a helper that generates a default set of validation tasks to run. We will eventually also have `validate_dimensions`, `validate_measures`, `validate_identifiers`, `validate_metrics`. All of these will essentially be two line functions (composing exposed `gen_<type>_tasks` on the `DataWarehouseTaskBuilder` and `run_tasks`). The helpers listed, `validate_data_sources` and the future ones, will be called by the CLI and likely Transform's internal servers. Others might call them, but it's more likely they'll compose a huge list of tasks using the different `gen_<type>_tasks` and their own custom tasks, and then running that list through `run_tasks`.